コマンドの補完確定時にまれに落ちる

- `v2.2.0` で tview のバージョンを上げたあたりからたまに起きる
- 補完候補が1つしか出ていない状態で起きる?
  - 起動 -> `:h` -> Enter -> Enter が一番再現率高そう

### Log

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x13d627c]

goroutine 1 [running]:
github.com/rivo/tview.(*Application).Run.func1()
        /home/runner/go/pkg/mod/github.com/rivo/tview@v0.0.0-20230104153304-892d1a2eb0da/application.go:244 +0x4d
panic({0x14847c0, 0x18c5cd0})
        /opt/hostedtoolcache/go/1.19.5/x64/src/runtime/panic.go:884 +0x212
github.com/rivo/tview.(*List).InsertItem(0x0, 0xffffffffffffffff, {0x14f65c9, 0x4}, {0x0, 0x0}, 0x0, 0x0)
        /home/runner/go/pkg/mod/github.com/rivo/tview@v0.0.0-20230104153304-892d1a2eb0da/list.go:357 +0xfc
github.com/rivo/tview.(*List).AddItem(...)
        /home/runner/go/pkg/mod/github.com/rivo/tview@v0.0.0-20230104153304-892d1a2eb0da/list.go:321
github.com/rivo/tview.(*InputField).Autocomplete(0xc000210b40)
        /home/runner/go/pkg/mod/github.com/rivo/tview@v0.0.0-20230104153304-892d1a2eb0da/inputfield.go:355 +0x53c
github.com/rivo/tview.(*InputField).InputHandler.func1.1()
        /home/runner/go/pkg/mod/github.com/rivo/tview@v0.0.0-20230104153304-892d1a2eb0da/inputfield.go:559 +0x48
github.com/rivo/tview.(*InputField).InputHandler.func1(0xc0000da020, 0x13d2c7f?)
        /home/runner/go/pkg/mod/github.com/rivo/tview@v0.0.0-20230104153304-892d1a2eb0da/inputfield.go:730 +0xacd
github.com/rivo/tview.(*Box).WrapInputHandler.func1(0xc000210b40?, 0x10?)
        /home/runner/go/pkg/mod/github.com/rivo/tview@v0.0.0-20230104153304-892d1a2eb0da/box.go:167 +0x53
github.com/rivo/tview.(*Grid).InputHandler.func1(0xc0000da020, 0x10?)
        /home/runner/go/pkg/mod/github.com/rivo/tview@v0.0.0-20230104153304-892d1a2eb0da/grid.go:275 +0x1aa
github.com/rivo/tview.(*Box).WrapInputHandler.func1(0xc00037b320?, 0xc00028fe38?)
        /home/runner/go/pkg/mod/github.com/rivo/tview@v0.0.0-20230104153304-892d1a2eb0da/box.go:167 +0x53
github.com/rivo/tview.(*Application).Run(0xc000176000)
        /home/runner/go/pkg/mod/github.com/rivo/tview@v0.0.0-20230104153304-892d1a2eb0da/application.go:336 +0x77d
github.com/arrow2nd/nekome/v2/app.(*App).Run(0xc0000a5560)
        /home/runner/work/nekome/nekome/app/app.go:311 +0xc6
main.main()
        /home/runner/work/nekome/nekome/main.go:15 +0x47
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -2,6 +2,12 @@
 
 ## [Unreleased]
 
+## [v2.2.2] - 2023-02-01
+
+### Fixed
+
+- コマンドの補完確定時にまれに落ちる
+
 ## [v2.2.1] - 2023-01-23
 
 ### Fixed
@@ -202,7 +208,8 @@
 
 - リリースしました！ 😸
 
-[unreleased]: https://github.com/arrow2nd/nekome/compare/v2.2.1...HEAD
+[unreleased]: https://github.com/arrow2nd/nekome/compare/v2.2.2...HEAD
+[v2.2.2]: https://github.com/arrow2nd/nekome/compare/v2.2.1...v2.2.2
 [v2.2.1]: https://github.com/arrow2nd/nekome/compare/v2.2.0...v2.2.1
 [v2.2.0]: https://github.com/arrow2nd/nekome/compare/v2.1.0...v2.2.0
 [v2.1.0]: https://github.com/arrow2nd/nekome/compare/v2.0.10...v2.1.0

--- a/app/app.go
+++ b/app/app.go
@@ -277,7 +277,6 @@ func (a *App) initAutocomplate() {
 	if err := a.commandLine.SetAutocompleteItems(cmds); err != nil {
 		shared.SetErrorStatus("Init - CommandLine", err.Error())
 	}
-
 }
 
 // execStartupCommands : 起動時に実行するコマンドを一括で実行

--- a/app/commandline.go
+++ b/app/commandline.go
@@ -121,11 +121,11 @@ func (c *commandLine) getAutocompleteItems(currentText string) []string {
 	c.isAutocompleteDisplaying = true
 
 	if currentText == "" {
-		// 入力中でないなら何も返さない
-		if c.inputField.GetLabel() == "" {
-			return nil
+		// 入力中の場合のみ全ての候補を返す
+		if c.inputField.GetLabel() == ":" {
+			return c.autocomplateItems
 		}
-		return c.autocomplateItems
+		return nil
 	}
 
 	for _, cmd := range c.autocomplateItems {

--- a/app/commandline.go
+++ b/app/commandline.go
@@ -21,7 +21,7 @@ type commandLine struct {
 func newCommandLine() *commandLine {
 	return &commandLine{
 		inputField:               tview.NewInputField(),
-		autocomplateItems:        []string{},
+		autocomplateItems:        nil,
 		isAutocompleteDisplaying: false,
 		backspaceCount:           0,
 	}
@@ -86,7 +86,9 @@ func (c *commandLine) SetAutocompleteItems(cmds []string) error {
 		c.autocomplateItems = append(c.autocomplateItems, cmd)
 	}
 
-	sort.Slice(c.autocomplateItems, func(i, j int) bool { return c.autocomplateItems[i] < c.autocomplateItems[j] })
+	sort.Slice(c.autocomplateItems, func(i, j int) bool {
+		return c.autocomplateItems[i] < c.autocomplateItems[j]
+	})
 
 	return nil
 }
@@ -119,6 +121,10 @@ func (c *commandLine) getAutocompleteItems(currentText string) []string {
 	c.isAutocompleteDisplaying = true
 
 	if currentText == "" {
+		// 入力中でないなら何も返さない
+		if c.inputField.GetLabel() == "" {
+			return nil
+		}
 		return c.autocomplateItems
 	}
 

--- a/go.mod
+++ b/go.mod
@@ -29,7 +29,7 @@ require (
 require (
 	github.com/gdamore/tcell/v2 v2.5.4
 	github.com/pkg/browser v0.0.0-20210911075715-681adbf594b8
-	github.com/rivo/tview v0.0.0-20230104153304-892d1a2eb0da
+	github.com/rivo/tview v0.0.0-20230130130022-4a1b7a76c01c
 	github.com/stretchr/testify v1.8.1
 	golang.org/x/crypto v0.5.0
 	golang.org/x/sync v0.1.0

